### PR TITLE
Adding extract_subset_granule

### LINKED ISSUE #82: Downloading subsetted dataset from PO.DAAC
- This issue is raised to track the issue raised in the PO.DAAC forum [here](https://podaac.jpl.nasa.gov/forum/viewtopic.php?f=53&t=424).

### PR DESCRIPTION
**Pull request by Omkar20895**
- This function iterates continuously for the given job token and downloads it as soon as the job is done.
- I am facing an issue after downloading and extracting the zip file 

> error : zipfile.BadZipfile: File is not a zip file

I tried to remove this error but couldn't I would like to use some fresh perspective. Pardon me if the error was caused by something silly. Ok Thanks. :) 

this closes #82

### CODE REVIEW

**Comment by lewismc** [comment]
Hi @Omkar20895 can you please shift this new function into the granule_subset function rather than creating yet another function? The idea is to keep the core Podaacpy functions exactly as specified within the Podaac Webservices API Specification. The core fucntion names can be seen on the README.
If you could please update the PR then i will check it out and also implement the test function.
Thank you v much for submitting ht patch Omkar.

**Comment by Omkar20895** [comment]
@lewismc sure thing!

**Comment by Omkar20895** [comment]
Done! Don't mind the commits, I will squash them at the end.

File: podaac/podaac.py [hunk 1]
```diff
@@ -501,7 +507,28 @@ def granule_subset(self, input_file_path):
...
+                flag = 1
+            if (status == "error"):
+                raise Exception(
+                    "Unexpected error occured for the subset job you have requested")
+            time.sleep(1)
+
+        print("Done! downloading the dataset zip .....")
+        download_url = subset_response_json['resultURLs'][0]
+        if path == '':
+            path = os.path.join(os.path.dirname(__file__), 'dataset.zip')
```
Comment (lewismc) [inline podaac/podaac.py]: Please grab the dataset name correctly from the ```download_url``` and append the .zip to this name.

**Comment by Omkar20895** [comment]
@lewismc Done! Apparently I did a silly mistake in my code, it works fine now. Thanks. :)

File: podaac/podaac.py [hunk 2]
```diff
@@ -501,7 +507,32 @@ def granule_subset(self, input_file_path):
...
+        split = download_url.split('/')
+        length = len(split)
+        zip_file_name = split[length-1]
+        if path == '':
+            path = os.path.join(os.path.dirname(__file__), zip_file_name)
+        else:
+            path = path + zip_file_name
+        response = urlretrieve(download_url, path)
+        zip_content = zipfile.ZipFile(path)
+        z.extractall()
```
Comment (lewismc) [inline podaac/podaac.py]: Veriable Z does not exist yet. Also I wonder if we can create a small unit test? This is looking great BTW @Omkar20895 .

**Comment by Omkar20895** [comment]
oh shoot! pardon me, I don't know what's wrong with me 😅  I will update it with a unit test and everything should be fine then. Thanks.

**Comment by lewismc** [comment]
+1 Omkar please merge with master. We can release podaacpy 1.5.0 and make the upgrade over in OCW. this is the best functionality yet in podaacpy thank you for working on this.

**Comment by Omkar20895** [comment]
No problems. Thank you very much! :)

**Pull request merged**